Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMElementGtk.cpp

```diff
@@ -819,22 +819,6 @@ void webkit_dom_element_scroll_into_view_if_needed(WebKitDOMElement* self, gbool
     item->scrollIntoViewIfNeeded(centerIfNeeded);
 }
 
-void webkit_dom_element_scroll_by_lines(WebKitDOMElement* self, glong lines)
-{
-    WebCore::JSMainThreadNullState state;
-    g_return_if_fail(WEBKIT_DOM_IS_ELEMENT(self));
-    WebCore::Element* item = WebKit::core(self);
-    item->scrollByLines(lines);
-}
-
-void webkit_dom_element_scroll_by_pages(WebKitDOMElement* self, glong pages)
-{
-    WebCore::JSMainThreadNullState state;
-    g_return_if_fail(WEBKIT_DOM_IS_ELEMENT(self));
-    WebCore::Element* item = WebKit::core(self);
-    item->scrollByPages(pages);
-}
```

Comment:
Hi, these are public APIs that can never be removed, but they're probably unused so it's probably safe to remove the implementations and just leave the functions empty.

Reply:
@mcatanzaro to be extra-sure, you are proposing the code to be something along this?

```cpp
// Element.scrollByLines and Element.scrollByPages have been removed. On the other hand, 
// these two functions are Public API and shall not be removed. 
// see https://bugs.webkit.org/show_bug.cgi?id=264732
void webkit_dom_element_scroll_by_lines(WebKitDOMElement* self, glong lines) {}
void webkit_dom_element_scroll_by_pages(WebKitDOMElement* self, glong pages) {}
```

Reply:
Right, but you don't need to add comments because the entire file is public API.

I would do:

```
void webkit_dom_element_scroll_by_lines(WebKitDOMElement* self, glong lines)
{
}

void webkit_dom_element_scroll_by_pages(WebKitDOMElement* self, glong pages)
{
}
```

Reply:
Basically at this point it's an established pattern that DOM APIs can get removed/broken at any time. They are all deprecated and have already been removed in the newer API versions, and most likely no applications actually use them, so it's sort of OK for them to break. But we cannot ever remove public symbols from the library, so those stick around even though they won't do anything anymore.

Reply:
done. Thanks.